app/controllers/administrative-units/administrative-unit/governing-bodies/governing-body/mandatory/edit.js

```diff
@@ -0,0 +1,54 @@
+import Controller from '@ember/controller';
+import { dropTask } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { combineFullAddress } from 'frontend-contact-hub/models/address';
+
+export default class AdministrativeUnitsAdministrativeUnitGoverningBodiesGoverningBodyMandatoryEditController extends Controller {
+  @service router;
+
+  @tracked isCurrentPositionCheck;
+
+  setup() {
+    this.isCurrentPositionCheck = this.isCurrentPosition;
+  }
+
+  get isCurrentPosition() {
+    return this.mandatory.endDate !== null;
+  }
+
+  get mandatory() {
+    return this.model.mandatory;
+  }
+
+  get administrativeUnit() {
+    return this.model.administrativeUnit;
+  }
+
+  get governingBody() {
+    return this.model.governingBody;
+  }
+
+  @dropTask
+  *save(event) {
+    event.preventDefault();
+
+    let contacts = yield this.mandatory.contacts;
+    if (contacts.firstObject.hasDirtyAttributes) {
+      let address = yield contacts.firstObject.contactAddress;
+
+      if (address && address.hasDirtyAttributes) {
```

Comment:
I think this needs to happen regardless if the contacts has changes, right? If the user only edits the address fields but not the contact info, this wouldn't be saved at the moment I think?

Reply:
Ah, I thought that as an address is part of the contact, this can indicate a change.

Reply:
yea changes in relationships will not mark this flag as dirty. There is no easy "relationship" dirtiness tracking in ember data :disappointed: